src/transformers/tokenization_utils.py

```diff
@@ -103,44 +123,100 @@ def truncate_and_pad(
-        - ``pretrained_init_configuration``: a python ``dict`` with, as keys, the `short-cut-names` (string) of the pretrained models, and as associated values, a dictionnary of specific arguments to pass to the ``__init__``method of the tokenizer class for this pretrained model when loading the tokenizer with the ``from_pretrained()`` method.
+        if isinstance(encoding, Encoding):
+            encoding = [encoding]
 
-    Parameters:
+        self._encodings = encoding
 
-        - ``bos_token``: (`Optional`) string: a beginning of sentence token. Will be associated to ``self.bos_token`` and ``self.bos_token_id``
+    def __getitem__(self, item: Union[int, str]) -> Encoding:
+        if isinstance(item, str):
+            return self.data[item]
+        elif self._encodings is not None:
+            return self._encodings[item]
+        else:
+            raise KeyError("int index is supported only on {} from a Rust tokenizer".format(type(self).__name__))
 
-        - ``eos_token``: (`Optional`) string: an end of sentence token. Will be associated to ``self.eos_token`` and ``self.eos_token_id``
+    def __getattr__(self, item: str):
+        return self.data[item]
 
-        - ``unk_token``: (`Optional`) string: an unknown token. Will be associated to ``self.unk_token`` and ``self.unk_token_id``
+    @property
+    def encodings(self) -> Optional[List[Encoding]]:
+        """
+        Return the list all encoding from the tokenization process
+        :return: List[Encoding] or None if input was tokenized through Python tokenizer
+        """
+        return self._encodings
 
-        - ``sep_token``: (`Optional`) string: a separation token (e.g. to separate context and query in an input sequence). Will be associated to ``self.sep_token`` and ``self.sep_token_id``
+    def keys(self):
+        return self.data.keys()
 
-        - ``pad_token``: (`Optional`) string: a padding token. Will be associated to ``self.pad_token`` and ``self.pad_token_id``
+    def values(self):
+        return self.data.values()
 
-        - ``cls_token``: (`Optional`) string: a classification token (e.g. to extract a summary of an input sequence leveraging self-attention along the full depth of the model). Will be associated to ``self.cls_token`` and ``self.cls_token_id``
+    def items(self):
+        return self.data.items()
 
-        - ``mask_token``: (`Optional`) string: a masking token (e.g. when training a model with masked-language modeling). Will be associated to ``self.mask_token`` and ``self.mask_token_id``
+    def char_to_token_offsets(self, sentence: int, char: int) -> Tuple[int, int]:
+        """
+        Find the Offsets of the token containing the character at the specified position
+        :param sentence: Index of the sentence relative to the batch provided to the tokenizer.
+        :param char: Char index to get the relative token offsets
+        :return: (token start, token end)
```

Comment:
(Applicable to most other docstrings) We use [Google style doc](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) in the library - could we try and use them here as well?